activemodel/lib/active_model/errors.rb

```diff
@@ -453,6 +460,19 @@ def normalize_message(attribute, message, options)
         message
       end
     end
+
+    def normalize_code(attribute, message, options)
+      code = case message
+      when Symbol
+        message
+      when Proc
+        message.call.to_sym
+      else
+        message.to_sym
+      end
+
+      { code: code }.merge(options.except(*CALLBACKS_OPTIONS + MESSAGE_OPTIONS))
```

Comment:
Ah, we also return other options. Got it. So I'd not call this method `codes`. But I can't think in a better name right now.